lib/cucumber/core/gherkin/parser.rb

```diff
@@ -18,19 +18,8 @@ def initialize(receiver, event_bus, gherkin_query)
         end
 
         def document(document)
-          messages = ::Gherkin.from_source(document.uri, document.body, gherkin_options(document))
-          messages.each do |message|
-            event_bus.envelope(message)
-            gherkin_query.update(message)
-            if !message.gherkin_document.nil?
-              event_bus.gherkin_source_parsed(message.gherkin_document)
-            elsif !message.pickle.nil?
-              receiver.pickle(message.pickle)
-            elsif message.parse_error
-              raise Core::Gherkin::ParseError.new("#{document.uri}: #{message.parse_error.message}")
-            else
-              raise "Unknown message: #{message.to_hash}"
-            end
+          source_messages(document).each do |message|
+            store_metadata(message, document)
```

Comment:
```suggestion
            process(message, document)
```

Reply:
I think `store_metadata` is a bit specific - we do a bunch of things in this message, but in general we're processing that message, so I think that's a better name.